src/HydraAuction/OnChain/FeeEscrow.hs

```diff
@@ -23,12 +23,15 @@ import HydraAuctionUtils.Plutus (
   byAddress,
   nothingForged,
  )
+import HydraAuctionUtils.Types.Natural (naturalToInt)
 
 {-# INLINEABLE mkFeeEscrowValidator #-}
 mkFeeEscrowValidator :: AuctionTerms -> FeeEscrowDatum -> FeeEscrowRedeemer -> ScriptContext -> Bool
 mkFeeEscrowValidator terms () DistributeFees context =
-  -- Every delegate is payed at least the expected proportion of fee
-  all receivesProportionOfFee (delegates terms)
+  -- There is one input spent from the fee escrow validator with enough ADA to pay the fees
+  adaValueOf singleFeeInputValue >= expectedFeePerDelegeate * length (delegates terms)
```

Comment:
But if all delegates get their fee, then everything is okay by construction, no?